### PR TITLE
Feat/442 perpetual open positions

### PR DESCRIPTION
**Pull request by ShynRou**
Resolves https://taiga.sovryn.app/project/sovryn-light/task/442

Moved perpUtils fully into the project and added the required dependency for liquidation approximations.
Implemented missing calculations in EditPositionSizeDialog and EditLeverageDialog, still missing review and excecution though.
Updated TradeForm to use mid price.
Implemented most of AmmDepthChart, except for latest trade price and trend.

### CODE REVIEW

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/app/pages/PerpetualPage/components/AmmDepthChart/index.tsx [hunk 1]
```diff
@@ -77,7 +76,7 @@ export const AmmDepthChart: React.FC<AmmDepthChartProps> = ({ pair }) => {
           <>
             {data?.shorts?.map((entry, index) => (
               <AmmDepthChartRow
-                key={entry.price}
+                key={entry.id}
```
[the remark this answers is not in the document]
Comment (ShynRou) [reply src/app/pages/PerpetualPage/components/AmmDepthChart/index.tsx:79]: Prices as id don't work while the hook is still fetching data and all prices are 0, that's why I changed it. The key is only really important for animations or when remounting of components is necessary.

File: src/app/pages/PerpetualPage/components/OpenPositionsTable/OpenPositionRow.tsx [hunk 2]
```diff
@@ -79,73 +89,94 @@ export const OpenPositionRow: React.FC<OpenPositionRowProps> = ({ item }) => {
...
-          value={item.realized.shortValue}
-          assetString={pair.shortAsset}
-        />
-        <AssetValue
-          className="tw-block"
-          value={item.realized.longValue}
-          assetString={pair.longAsset}
-          isApproximation
-        />
+        {item.realized && (
```
[the remark this answers is not in the document]
Comment (ShynRou) [reply src/app/pages/PerpetualPage/components/OpenPositionsTable/OpenPositionRow.tsx:163]: As I remember it vasili wanted to work out with backend how to calculate the realized value.
Realized makes sense if you for example make profit ant then decrease the size and take out margin.

File: src/app/pages/PerpetualPage/components/NewPositionCard/components/SlippageFormStep.tsx [hunk 3]
```diff
@@ -44,17 +55,36 @@ export const SlippageFormStep: TransitionStep<NewPositionCardStep> = ({
...
+      perpParameters,
+    );
+    return calculateApproxLiquidationPrice(
+      amount,
+      margin,
+      ammState,
+      perpParameters,
+    );
+  }, [
+    trade.amount,
```
[the remark this answers is not in the document]
Comment (ShynRou) [reply src/app/pages/PerpetualPage/components/NewPositionCard/components/SlippageFormStep.tsx:81]: Well it's the min liquidation price, which isn't affected by slippage. The real liquidation price of cause would change depending on real slippage, but not the minimum as it should reflect 0% slippage.

File: src/app/pages/PerpetualPage/components/AmmDepthChart/index.tsx [hunk 4]
```diff
@@ -156,7 +155,7 @@ export const AmmDepthChart: React.FC<AmmDepthChartProps> = ({ pair }) => {
             </tr>
             {data?.longs?.map((entry, index) => (
               <AmmDepthChartRow
-                key={entry.price}
+                key={entry.id}
```
[the remark this answers is not in the document]
Comment (ShynRou) [reply src/app/pages/PerpetualPage/components/AmmDepthChart/index.tsx:158]: See above

File: src/app/pages/PerpetualPage/components/OpenPositionsTable/OpenPositionRow.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (tiltom) [reply src/app/pages/PerpetualPage/components/OpenPositionsTable/OpenPositionRow.tsx:163]: Yeah it makes sense, he also wrote it to the perpetuals chat in the meanwhile.

File: src/app/pages/PerpetualPage/components/NewPositionCard/components/SlippageFormStep.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (tiltom) [reply src/app/pages/PerpetualPage/components/NewPositionCard/components/SlippageFormStep.tsx:81]: Sorry, my bad :D

**Review by tiltom — APPROVED** [review]

**Pull request merged**